docs/misc/localization.md

```diff
@@ -0,0 +1,107 @@
+# Localization
+
+Suite uses `react-intl` package for all in-app localization needs.
+Definitions of all messages are stored in [messages.ts](https://github.com/trezor/trezor-suite/blob/develop/packages/suite/src/support/messages.ts).
+
+To allow non-developers to improve these messages through user-friendly interface, we upload them to Crowdin via [trezor-translations-manager](https://github.com/trezor/trezor-suite/tree/develop/packages/translations-manager) (TTM) which takes care of conversion to CSV file format before the upload. *Be aware that TTM is rather legacy project and most likely could be replaced with `formatjs` cli.*
+
+After strings have been translated we use TTM again to download the messages and generate language json files. They are automatically copied to [suite-data package](https://github.com/trezor/trezor-suite/tree/develop/packages/suite-data/files/translations).
+To finish the process these files need to be commited to the repository.
+
+## Message definitions
+[messages.ts](https://github.com/trezor/trezor-suite/blob/develop/packages/suite/src/support/messages.ts) is the place where you add new messages to be used in Suite. It's basically just a huge object where a key is an ID of the message and a value is the message definition.
+
+### Structure
+
+- `id`: We don't have strict conventions for generating these IDs, although using a prefix `TR_`, or expanded variant `TR_<SCOPE>`, where scope is, for example, "ONBOARDING" is really handy. ID must be the same as the object's key.
```

Comment:
Cause you need to refactor the tooling (and `Translation` component and `useTranslation` hook to not rely on id field) and no one did that yet. 
Official shape of MessageDescriptor object for `react-intl` is
```
export interface MessageDescriptor {
    id?: string | number;
    description?: string | object;
    defaultMessage?: string | MessageFormatElement[];
}
```
I'll add to list of things to do along TTM refactor, but I don't like to change too many things to our liking cause it will prevent us from using standardised tools and someone will need to maintain it :)

Reply:
> I'll add to list of things to do along TTM refactor (#3583)

Perfect. Resolving.

>  I don't like to change too many things to our liking cause it will prevent us from using standardised tools and someone will need to maintain it :)

Agree that our shim on top of react-intl should be as thin as possible. At the very minimum we can *just* do this
```diff
export interface ExtendedMessageDescriptor extends MessageDescriptor {
-    id: keyof typeof messages; 
+    id: typeof messages[keyof typeof messages]['id'];
```

which would make us refer directly to the ids. The keys would still exist but wouldn't be taken into account, which is not ideal either.

Btw. the react-intl docs generaly discourage from our approach of defining the messages before-hand at one place. [They recommend](https://formatjs.io/docs/getting-started/message-declaration) defining the messages always in-line. .) 

I digress. We can talk about this in the linked issue.

Reply:
>Btw. the react-intl docs generaly discourage from our approach of defining the messages before-hand at one place. They recommend defining the messages always in-line. .)

We've been there, later we switched for single file which has its own pain points, but in the end it works better for us.